### ISSUE TITLE
Drop the default test users

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#1138]
# Task Description

We have some test users (`deployer`, `developer01`, `maven`, `user`) added to a production resource (`strongbox-security/strongbox-user-management/src/main/resources/etc/conf/strongbox-security-users.xml`; please, note that with the implementation of #965, the `xml` extension will become `yaml`). We need to remove the `developer01`, `maven`, `user` these from there. When tests need these users, they should create them.

The `deployer` user should be the only one left and it should have its access model dropped. It should be configured to have read and write access to all storages and repositories.

This task will most-likely break the `strongbox-web-integration-tests`. The `ArtifactClient` and `RestClient` classes will need to be updated accordingly.

# Acceptance Test

* [ ] Building the code with `mvn clean install -Dintegration.tests` still works.
* [ ] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Task Links

Depends on: #965.
Documentation follow-up task: #1094.

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

**Comment by evrimnurcelik** [comment]
can you please assign tis one to me? I'm a first timer trying to start with contributing to open source projects.

**Comment by carlspring** [comment]
@evrimnurcelik ,

Feel free to give it a go, as well as to join our 
[chat channel](https://chat.carlspring.org/)! :)
